### DISCUSSION TITLE
PDF 属性情報のタイトルを設定したい

### DISCUSSION THREAD

**Discussion by bsdmad** [thinreports/thinreports#37] [Q&A] [answered]
とても便利に使わせていただいています。ありがとうございます。

現在、ver. 0.12.1 の thinreports-generator gem を使っています。
PDF ファイルの属性情報のタイトルに任意の文言を設定したいのですが、以下の箇所を見る限り
それはできないように思います。

https://github.com/thinreports/thinreports-generator/blob/main/lib/thinreports/basic_report/generator/pdf.rb#L18

理由があってのことかと思いますが、設定できるように変更していただけないものでしょうか？

追伸：
title 指定箇所は見つけられたのですが、これを指定できるようにするにはどうすれば良いかが
分かりませんでした。pull request できず、申し訳ありません。

**Comment by hidakatsuya** [comment, marked as answer]
その `title` は、Editor のレイアウト設定のページタイトルに設定した内容が反映されます。
https://github.com/thinreports/thinreports/blob/main/getting-started/basic-format.md#creating-a-new-layout

コードから設定したいということであれば、現状 API はありません。ただ、何か制約があって実装できないわけではありません。

**Reply by bsdmad** [reply]
回答、ありがとうございます。

会員証や請求書などを PDF 生成しており、ユーザ個別の情報（◯◯様 etc.）を埋め込みたいと
思いました。自分でもコードを拝見して実装できないか挑戦してみたいと思いますが、可能で
あれば今後、実装をご検討いただければ幸いです。

**Reply by hidakatsuya** [reply]
こんな感じで実装しようと思いますがいかがでしょう？
https://github.com/thinreports/thinreports-generator/pull/125

**Reply by hidakatsuya** [reply]
v0.13.0 としてリリースしました。お試しください。
https://github.com/thinreports/thinreports-generator/blob/main/CHANGELOG.md#0130

**Reply by bsdmad** [reply]
ご対応、ありがとうございます。週末になってしまうと思いますが、利用させていただきますm(_ _)m

**Comment by bsdmad** [comment]
v0.13.0 を試してみました。例示いただいた書式でタイトルが指定できることを確認しました。
ただ、以下の書式だと「unknown keyword: :title」のエラーになりました。今、この書式は
非推奨でしょうか？

```ruby
Thinreports::Report.generate filename: @file, title: 'コレはテストです', security: security do |report|
  （省略）
end
```

**Reply by hidakatsuya** [reply]
いえ、対応漏れです。v0.13.1 で修正します。

**Reply by bsdmad** [reply]
早々のご確認、ありがとうございます。よろしくお願いします。

**Reply by hidakatsuya** [reply]
v0.13.1 で修正しました。
https://github.com/thinreports/thinreports-generator/pull/127

**Reply by bsdmad** [reply]
想定通りのタイトルが設定できることを確認しました。有難うございました！

なお余談ですが Prawn のバグで security を設定すると日本語のタイトルは文字化けしてしました。
こちらはもう少し調べてから Prawn の方で issue を登録してみようと思います。

**Comment by bsdmad** [comment]
ご対応、ありがとうございました。

※私がcloseして良いか分かりませんので、間違えていたらお手数ですがご容赦ください。

**Reply by hidakatsuya** [reply]
close ありがとうございます。全く問題ありません 👍🏻
